biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -169,6 +176,32 @@ public static int run(String args[]) throws Throwable {
 		return target.launch(args);
 	}
 
+	/**
+	 * Launch with optional properties & optional bundle activator. If the
+	 * properties are null, they are attempted to be read from
+	 * `launcher.properties`.
+	 *
+	 * @param args
+	 * @param propertiesOrNull the properties or null
+	 * @param activator null, or a Bundle Activator that will be called back
+	 *            before the bundles are started, see
+	 *            {@link EmbeddedActivatorPhase#BEFORE_BUNDLES_START}.
+	 * @return the exit code
+	 * @throws Throwable
+	 */
+	public static int launch(String[] args, Properties propertiesOrNull, BundleActivator activator)
```

Comment:
We have `propertiesOrNull` but not `activatorOrNull`? I am not sure I like the `OrNull` variable name suffix, but we should at least be consistent.